### ISSUE TITLE
Fragments with a function-style defaultValue error in v6

### ISSUE THREAD

**Issue by grayt0r** [adopted-ember-addons/ember-data-model-fragments#470]
Versions:
* `ember-data-model-fragments`: `v6.0.1`
* `ember-data`: `3.28.13`

When pushing a payload to the store for a model that contains a fragment with a function-style `defaultValue` I receive the following error:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at -private.js:2170:1
    at changeProperties (index.js:589:1)
    at ExampleModel._notifyProperties (-private.js:2168:1)
    at InternalModel.setupData (-private.js:5991:1)
    at Store._load (-private.js:9153:1)
    at Store._pushInternalModel (-private.js:9380:1)
    at -private.js:9331:1
    at Backburner._run (backburner.js:843:1)
    at Backburner._join (backburner.js:822:1)
    at Backburner.join (backburner.js:632:1)
```

The model is defined like this:

```js
export default class ExampleModel extends Model {
  @attr name;

  @fragment('example-fragment', { defaultValue: () => ({}) }) exampleFragment;
}
```

The error goes away and everything works when I downgrade to `ember-data-model-fragments` `v5.0.0` and `ember-data` `v3.27.1`.

I can also fix the error by swapping to a non-function `defaultValue`, i.e.:

```js
export default class ExampleModel extends Model {
  @attr name;

  @fragment('example-fragment', { defaultValue: {} }) exampleFragment;
}
```

Is this a known limitation of `v6`? Or a bug?

A minimal reproduction can be found here: https://github.com/grayt0r/ember-data-model-fragments-bug ([relevant commit](https://github.com/grayt0r/ember-data-model-fragments-bug/commit/ccfc8fb5608a9a09cd323add8856f322368b2c6b)).

Thanks!

**Comment by VincentMolinie** [comment]
Oh I actually got the exact same issue ! This PR might fix your issue (https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/474)

**Comment by grayt0r** [comment]
@VincentMolinie looks like it does... thank you 🙌

**Comment by knownasilya** [comment]
Released as v6.0.2
